src/Randousel/randousel.svelte

```diff
@@ -0,0 +1,99 @@
+<script>
+  import { Card, CardSubtitle, CardTitle, CardText, CardActions, Button, MaterialApp } from 'svelte-materialify';
+  import { format } from 'd3';
+
+  export let location;
+  export let wtfData;
+  let count = 0;
+  let data = []; // Initialize data
+  let selectedIndex = 0; // Randomly pick an array element
+  let selectedWTF;
+
+  const loading = Promise.all([location, wtfData])
+    .then( ([locationResponse, wtfResponse]) => {
+      data = wtfResponse['data']['requests'].filter(d => d.wtf);
+      console.log(data)
+      selectedIndex = data.findIndex(d => d.state === locationResponse.region);
+    });
+
+  const amountFormatter = format('.2s');
+
+  function titleCase(str) {
+    str = str.toLowerCase().split(' ');
+    for (var i = 0; i < str.length; i++) {
+      str[i] = str[i].charAt(0).toUpperCase() + str[i].slice(1); 
+    }
+    return str.join(' ');
+  }
+
+  function randomize() {
+    // Get a random index
+    const newIndex = Math.floor(Math.random() * data.length);
+
+    if (data.length > 1 && newIndex === selectedIndex) {
+      // If new index matches the current index, re-randomize;
+      // check array length to avoid infinite recursion
+      randomize();
+    } else {
+      // Else update index
+      selectedIndex = newIndex;
+    }
+  }
+	
+	function handleClick() {
+    if (count == 51) {
+      count = 1
+    } else {
+      count += 1
+    }
+	}
+
+  
+  $: if (count == 0) {
+    selectedWTF = data[selectedIndex]
+  } else {
+    selectedWTF = data.filter(d => +d.rank == count)[0]
+  }
+</script>
+
+<p>Here's an example of how local government spent the federal COVID-19 relief fund on the criminal justice system.</p>
```

Comment:
This text is static and outside of the dynamic parts of the component, so I'd probably suggest it belongs in the HTML itself.

Reply:
+++